File: JULY-2025/19-07-2025/p1233.rb
# @param {String[]} folder
# @return {String[]}
def remove_subfolders(folder)
    folder.sort!
    if folder.length == 0
        return folder
    end

    stack = [folder[0]]
    for i in 1...folder.length
        if folder[i].index(stack[stack.length-1] + "/") == 0
            i += 1
        else
            stack.push(folder[i])
        end
    end
    return stack
end

File: JULY-2025/19-07-2025/p1233.rs
use std::path::Path;

impl Solution {
    pub fn remove_subfolders(folder: Vec<String>) -> Vec<String> {
        let mut result: Vec<String> = Vec::new();
        let mut folder = folder;
        folder.sort();

        let mut curr = &folder[0];
        result.push(curr.clone());

        for f in folder.iter() {
            let path_curr = Path::new(curr);
            let path_f = Path::new(f);
            if path_f.starts_with(path_curr) {
                continue;
            } else {
                curr = &f;
                result.push(f.clone());
            }
        }

        result
    }
}